RFC/src/RFC-0120_Consensus.md

```diff
@@ -0,0 +1,212 @@
+* height;
+* prev_hash;
+* timestamp;
+* output_mr;
+* range_proof_mr;
+* kernel_mr;
+* total_kernel_offset;
+* total_difficulty;
+* nonce;
+* pow.
+
+The [block header] MUST confirm to the following:
+
+* The nonce and [PoW](#pow) must be a valid for the [block header].
+* All the merkle roots must be valid for the states after the [block] was applied to the local state.
+
+If the [block header] does not confirm to any of the above, the [block] needs to be rejected and the peer the it was received from, marked as a malicious peer.
+
+#### Version
+
+This is the version currently running on the chain.
+
+The version MUST confirm to the following:
+
+* It is represented as an unsigned 16-bit integer.
+* Version numbers should be incremented whenever there is a change in the blockchain schema starting from 1.
+
+#### Height
+
+A counter indicating how much blocks have passed since the genesis block (inclusive).
+
+The height MUST confirm to the following:
+
+* Represented as an unsigned 64-bit integer.
+* The height MUST be exactly 1 more than the block referenced in the `prev_hash` block header field.
+* The genesis block MUST have a height of 1.
+
+#### Prev_hash
+
+This is the hash of the previous block's header.
+
+The prev_hash MUST confirm to the following:
+
+* represented as an array of unsigned 8-bit integers (bytes).
+* MUST be a hash of the entire contents of the previous block's header.
+
+#### Timestamp
+
+This is the timestamp at which the block was mined.
+
+The timestamp MUST confirm to the following:
+
+* Must be transmitted as UNIX timestamp.
+* MUST be less than [FTL].
+* MUST be higher than the [MTP].
+
+#### Output_mr
+
+This is the merkle root of the outputs. This is calculated in the following way: Hash (txo MMR root  || roaring bitmap hash of UTXO indices).
+
+The output_mr MUST confirm to the following:
+
+* Represented as an array of unsigned 8-bit integers (bytes).
+* The hashing function used MUST be blake2b with a 256 bit digest.
+
+#### Range_proof_mr
+
+This is the merkle root of the range proofs.
+
+The range_proof_mr MUST confirm to the following:
+
+* Represented as an array of unsigned 8-bit integers (bytes).
+* The hashing function used must be blake2b with a 256 bit digest.
+
+#### Kernel_mr
+
+This is the merkle root of the outputs.
+
+The kernel_mr MUST confirm to the following:.
+
+* Must be transmitted as an array of unsigned 8-bit integers (bytes).
+* The hashing function used must be blake2b with a 256 bit digest.
+
+#### Total_kernel_offset
+
+This is total summed offset of all the transactions contained in this block.
+
+The total_kernel_offset MUST confirm to the following:
+
+* Must be transmitted as an array of unsigned 8-bit integers (bytes)
+
+#### Total_difficulty
+
+This is the total accumulated difficulty of the mined chained.
+
+The total_difficulty MUST confirm to the following:
+
+* Must be transmitted as unsigned 64-bit integer;
+* MUST be larger than the previous block's `total_difficulty`;
```

Comment:
* meet the difficulty target for the block as determined by the consensus difficulty algorithm